function/function_manager.rs

```diff
@@ -110,27 +124,73 @@ impl FunctionManager {
         let function_index = ReadThroughFunctionSignatureIndex::new(snapshot, self, buffered);
         // Translate to ensure the function calls are valid references. Type-inference is done at commit-time.
         Self::translate_functions(snapshot, &functions, &function_index)?;
-
-        for (function, definition) in zip(functions.iter(), definitions.iter()) {
+        for (function, definition) in zip(functions.iter(), definitions.clone()) {
             let index_key = NameToFunctionDefinitionIndex::build(function.name().as_str()).into_storage_key();
             let definition_key = &function.function_id;
             snapshot.put_val(index_key.into_owned_array(), ByteArray::copy(definition_key.bytes().bytes()));
             snapshot.put_val(
                 definition_key.clone().into_storage_key().into_owned_array(),
-                FunctionDefinition::build_ref(definition).into_bytes().into_array(),
+                FunctionDefinition::build_ref(definition.unparsed.as_str()).into_bytes().into_array(),
             );
         }
         Ok(functions)
     }
 
+    pub fn undefine_function(&self, snapshot: &mut impl WritableSnapshot, name: &str) -> Result<(), FunctionError> {
+        let definition_key = match self.get_function_key(snapshot, name) {
+            Err(source) => Err(FunctionError::FunctionRetrieval { source }),
+            Ok(None) => Err(FunctionError::FunctionNotFound {}),
+            Ok(Some(key)) => Ok(key),
+        }?;
+        snapshot.delete(definition_key.into_storage_key().into_owned_array());
```

Comment:
I you need a validation here to make sure it's not used in another function?

Reply:
Too much work to build out the infra for that if I need to switch to performance.
I have a BDD that makes sure the commit fails.

Reply:
leave a TODO then please

Reply:
It's already there.